### ISSUE TITLE
Failure to create azurerm_batch_job with multiple properties in common_environment_properties

### ISSUE THREAD

**Issue by mikevo13** [hashicorp/terraform-provider-azurerm#15647]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform (and AzureRM Provider) Version
Terraform 1.1.4
Azure RM 2.92.0

### Affected Resource(s)

* `azurerm_batch_job`

### Terraform Configuration Files

```hcl
resource "azurerm_resource_group" "example" {
  name     = "example-rg"
  location = "North Central US"
}

resource "azurerm_batch_account" "example" {
  name                = "exampleaccount1234"
  resource_group_name = azurerm_resource_group.example.name
  location            = azurerm_resource_group.example.location
}

resource "azurerm_batch_pool" "example" {
  name                = "examplepool"
  resource_group_name = azurerm_resource_group.example.name
  account_name        = azurerm_batch_account.example.name
  node_agent_sku_id   = "batch.node.ubuntu 18.04"
  vm_size             = "Standard_D2_v3"

  fixed_scale {
    target_dedicated_nodes = 1
  }

  storage_image_reference {
    publisher = "canonical"
    offer     = "ubuntuserver"
    sku       = "18.04-lts"
    version   = "latest"
  }
}

resource "azurerm_batch_job" "example" {
  name          = "examplejob"
  batch_pool_id = azurerm_batch_pool.example.id

  common_environment_properties = {
    a = "1"
    b = "2"
  }
}
```

### Expected Behaviour
Batch job created successfully

### Actual Behaviour

Encountered below error:
```
╷
│ Error: creating Job: (Name "examplejob" / Pool Name "examplepool" / Batch Account Name "exampleaccount1234" / Resource Group "example-rg"): batch.JobClient#Add: Failure responding to request: StatusCode=400 -- Original Error: autorest/azure: error response cannot be parsed: "{\r\n  \"odata.metadata\":\"https://exampleaccount1234.northcentralus.batch.azure.com/$metadata#Microsoft.Azure.Batch.Protocol.Entities.Container.errors/@Element\",\"code\":\"InvalidPropertyValue\",\"message\":{\r\n    \"lang\":\"en-US\",\"value\":\"The value provided for one of the properties in the request body is invalid.\\nRequestId:e7f4abed-a576-4c7a-85ea-a7fb77cc9db9\\nTime:2022-03-01T16:39:13.0743016Z\"\r\n  },\"values\":[\r\n    {\r\n      \"key\":\"PropertyName\",\"value\":\"name\"\r\n    },{\r\n      \"key\":\"PropertyValue\",\"value\":\"b\"\r\n    },{\r\n      \"key\":\"Reason\",\"value\":\"Duplicate Environment Setting name\"\r\n    }\r\n  ]\r\n}" error: json: cannot unmarshal object into Go struct field RequestError.Message of type string
│ 
│   with azurerm_batch_job.example,
│   on batch.tf line 71, in resource "azurerm_batch_job" "example":
│   71: resource "azurerm_batch_job" "example" {
│ 
│ creating Job: (Name "examplejob" / Pool Name "examplepool" / Batch Account
│ Name "exampleaccount1234" / Resource Group "example-rg"):
│ batch.JobClient#Add: Failure responding to request: StatusCode=400 --
│ Original Error: autorest/azure: error response cannot be parsed: "{\r\n
│ \"odata.metadata\":\"https://exampleaccount1234.northcentralus.batch.azure.com/$metadata#Microsoft.Azure.Batch.Protocol.Entities.Container.errors/@Element\",\"code\":\"InvalidPropertyValue\",\"message\":{\r\n
│ \"lang\":\"en-US\",\"value\":\"The value provided for one of the properties
│ in the request body is
│ invalid.\\nRequestId:e7f4abed-a576-4c7a-85ea-a7fb77cc9db9\\nTime:2022-03-01T16:39:13.0743016Z\"\r\n
│ },\"values\":[\r\n    {\r\n
│ \"key\":\"PropertyName\",\"value\":\"name\"\r\n    },{\r\n
│ \"key\":\"PropertyValue\",\"value\":\"b\"\r\n    },{\r\n
│ \"key\":\"Reason\",\"value\":\"Duplicate Environment Setting name\"\r\n
│ }\r\n  ]\r\n}" error: json: cannot unmarshal object into Go struct field
│ RequestError.Message of type string
╵
```

### Steps to Reproduce

1. `terraform apply`

### Important Factoids
If we keep one property only in common_environment_properties, everything work fine:

```hcl
resource "azurerm_batch_job" "example" {
  name          = "examplejob"
  batch_pool_id = azurerm_batch_pool.example.id

  common_environment_properties = {
    a = "1"
  }
}
```

**Comment by sinbai** [comment]
@mikevo13  thank you for reporting this issue. The [PR ](https://github.com/hashicorp/terraform-provider-azurerm/pull/15686)has been submitted to fix it.

**Comment by stephybun** [comment]
Closed by #15686
